### PR TITLE
feat: add dbt project indexer for fast model lookup

### LINKED ISSUE #210: feat: dbt project indexer for fast model lookup
## Problem

When users ask the agent lineage or model questions (e.g., _"What is the lineage of BASE_EMAIL__BREAD_SAVINGS?"_ or _"What dbt model creates the fact_page_impression table?"_), the agent has to **grep through 694+ raw SQL files** in `repos/`. This causes:

1. **Slow and unreliable searches** — scanning hundreds of files often returns incomplete results or times out
2. **Missed downstream dependencies** — the agent can find a model's `ref()` calls (upstream), but has no efficient way to find which other models reference it (downstream)
3. **No source-to-database mapping** — users asking _"What Snowflake table does `source('core', 'dim_offers')` point to?"_ have to manually correlate YAML source definitions with database schemas
4. **Repeated work** — every question re-scans the same files with no cached representation of the project structure

## Proposed Solution

A **backend-side dbt indexer** that parses all dbt SQL/YAML files in `repos/` and generates searchable markdown index files:

- **`manifest.md`** — one entry per model with path, materialization, refs (upstream), sources, and description. Sorted alphabetically for consistent grep results.
- **`sources.md`** — source-to-database mapping with schema and table names.

The agent reads a single `manifest.md` (~4000 lines for 695 models) instead of grepping through hundreds of individual SQL files.

### Use cases

| Use case | Before | After |
|----------|--------|-------|
| "What is the lineage of model X?" | Grep 694 SQL files, often miss downstream refs | Grep 1 file, find upstream + downstream in one search |
| "What creates table Y?" | Grep all files hoping the model name matches | Search manifest for model name, get path + materialization |
| "What source feeds into model Z?" | Read SQL, then find YAML source def | `sources` field inline in manifest |
| "Map source to Snowflake database" | Manually find YAML, parse database/schema | Grep `sources.md`, get database + schema + tables |
| "Show all incremental models" | Impossible without scanning every file | Grep manifest for `materialized: incremental` |

### Trigger points

| Trigger | When |
|---------|------|
| `nao sync` | After repos are cloned/pulled |
| FastAPI startup | When `nao chat` starts |
| `POST /api/refresh` | Manual or webhook trigger |
| Cron scheduler | `NAO_REFRESH_SCHEDULE` env var |

## Implementation

See PR #209 for the complete implementation.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### PR DESCRIPTION
**Pull request by ealexisaraujo**
## Summary

- **New dbt indexer** parses all SQL/YAML files in `repos/` and generates searchable markdown index files (`manifest.md` + `sources.md`) per dbt project
- **Hooks into `nao sync`** as a post-sync step, plus FastAPI startup, `/api/refresh`, and optional cron scheduler
- **Updates system prompt** to instruct the agent to search `dbt-index/` first before grepping raw SQL files
- **32 unit tests** covering SQL parsing, YAML parsing, project discovery, and error handling

## Motivation

When users asked the agent lineage or model questions (e.g., _"What is the lineage of stg_orders?"_ or _"What dbt model creates the fact_revenue table?"_), the agent had to **grep through hundreds of raw SQL files** in `repos/`. This caused several problems:

1. **Slow and unreliable** — searching hundreds of files often timed out or returned incomplete results
2. **Missed downstream dependencies** — the agent could find a model's `ref()` calls (upstream), but had no efficient way to find which other models reference it (downstream). It would have to grep every single SQL file for the model name
3. **No source-to-database mapping** — users asking _"What Snowflake table does `source('core', 'dim_offers')` point to?"_ had to manually correlate YAML source definitions with database schemas
4. **Repeated work** — every question re-scanned the same files. There was no cached representation of the project structure

### Use cases this enables

| Use case | Before | After |
|----------|--------|-------|
| "What is the lineage of model X?" | Grep hundreds of SQL files, often miss downstream refs | Grep 1 file (`manifest.md`), find upstream `refs` + downstream in one search |
| "What creates table Y?" | Grep all files hoping the model name matches | Search `manifest.md` for the model name, immediately get path + materialization |
| "What source feeds into model Z?" | Read the SQL file, then find the YAML source def | `sources` field shows `source_name.table_name` inline in manifest |
| "Map this source to a database table" | Manually find YAML, parse database/schema | Grep `sources.md` for the source name, get database + schema + tables |
| "Show me all incremental models" | Impossible without scanning every file | Grep `manifest.md` for `materialized: incremental` |

### Why index at the Python (FastAPI/CLI) layer

- FastAPI already owns the context lifecycle (refresh/startup)
- Python has mature YAML and regex parsing (`pyyaml` already a dependency)
- Output is file-based, consistent with the existing `databases/` pattern
- Both `nao sync` (CLI) and FastAPI (server) can import from `nao_core.dbt_indexer`

### Data flow

```
nao sync -p repositories
  │
  ├─ git clone/pull → repos/<name>/
  │
  └─ _index_dbt_projects()
       │
       ├─ find_dbt_projects()        ← scans repos/ for dbt_project.yml
       ├─ read_project_config()      ← reads project name, default materializations
       ├─ index_dbt_project()
       │    ├─ parse_yaml_sources()      ← YAML: source definitions
       │    ├─ parse_yaml_descriptions() ← YAML: model descriptions
       │    ├─ parse_sql_dependencies()  ← SQL: ref() and source() calls
       │    └─ parse_sql_config()        ← SQL: materialization config
       ├─ generate_manifest_md()     ← sorted alphabetically for grep
       └─ generate_sources_md()      ← source-to-database mapping
```

### Output (generated in user's project folder)

```
<project>/
├── dbt-index/
│   └── <repo-name>/
│       ├── manifest.md    (grep-friendly, one entry per model)
│       └── sources.md     (source-to-database mapping)
├── repos/                 (raw git repos — still needed for full SQL)
└── databases/             (database schema docs)
```

### Trigger points (4 ways to run)

| Trigger | When | Where |
|---------|------|-------|
| `nao sync` | After repos are cloned/pulled | `cli/nao_core/commands/sync/__init__.py` |
| FastAPI startup | When `nao chat` starts | `main.py` lifespan |
| `POST /api/refresh` | Manual or webhook trigger | `main.py` refresh endpoint |
| Cron scheduler | `NAO_REFRESH_SCHEDULE` env var | `main.py` APScheduler |

### manifest.md (grep-friendly)

```markdown
### stg_core__dim_products
- **path:** models/staging/core/stg_core__dim_products.sql
- **materialized:** view
- **refs:** —
- **sources:** core.dim_products

### int_orders__enriched
- **path:** models/transform/orders/int_orders__enriched.sql
- **materialized:** incremental
- **refs:** stg_core__dim_products, stg_core__fact_orders
- **sources:** —
```

The agent greps `manifest.md` for a model name and instantly finds:
- **Upstream:** `sources` and `refs` fields
- **Downstream:** grep for the model name in `refs:` lines of other models

## Files changed

| File | Change |
|------|--------|
| `cli/nao_core/dbt_indexer.py` | **New** — Core indexer implementation (single source of truth): SQL/YAML parsing, manifest/sources generation |
| `apps/backend/fastapi/dbt_indexer.py` | **New** — Re-export shim so FastAPI imports work |
| `apps/backend/fastapi/main.py` | Hook indexer into startup, refresh endpoint, and scheduler |
| `apps/backend/src/agents/user-rules.ts` | Add `indexed` field to Repository type, detect `dbt-index/` |
| `apps/backend/src/components/system-prompt.tsx` | Tell agent to search `dbt-index/` first; show indexed status |
| `cli/nao_core/commands/sync/__init__.py` | Run indexer after `nao sync -p repositories` |
| `apps/backend/fastapi/test_dbt_indexer.py` | **New** — 32 unit tests |

## Performance

- Tested on a real dbt project with **694 models + 129 YAML files** — indexed in **0.83 seconds**
- Handles edge cases: directories named `.sql`, binary files, Jinja in YAML, malformed YAML

## Test plan

- [x] `python -m pytest apps/backend/fastapi/test_dbt_indexer.py -v` — 32 tests pass
- [x] `make lint` (cli/) — ty, ruff check, ruff format all pass
- [x] `npm run lint` — TypeScript lint passes (0 errors)
- [x] Run `nao sync -p repositories` on real project — `dbt-index/` generated correctly
- [x] Run `nao chat` — agent searches `dbt-index/**/manifest.md` first (verified)
- [x] Test lineage question — agent finds upstream + downstream from manifest
- [x] `POST /api/refresh` — returns 200, triggers re-indexing
- [x] Verify `sources.md` contains correct source-to-database mappings

Fixes #210

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by Bl3f — COMMENTED** [review]
Hey, thank you so much for the contribution, to be honest this is great. There are a few things to change I think, a few part to refacto a bit or remove and the main question is around the manifest.md that might be a large file and how we should mitigate this in order to not overflow users context windows.

File: cli/nao_core/dbt_indexer.py [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+    database: str | None
+    schema_name: str
+    tables: list[str]
+
+
+# ---------------------------------------------------------------------------
+# Compiled regex patterns
+# ---------------------------------------------------------------------------
+
+RE_REF = re.compile(
```
Comment (Bl3f) [inline cli/nao_core/dbt_indexer.py:35]: As you tried it on a real dbt project I'm eager to see if this combination of regular expressions worked well, I'm a bit unsure to do it with regexp as dbt jinja syntax could be in some wild projects not compatible with this and we could miss a lot of lineage information this way (like generated join from macros, etc.)

File: cli/nao_core/commands/sync/__init__.py [hunk 2]
```diff
@@ -103,6 +119,9 @@ def sync(
             results.append(SyncResult.from_error(sync_provider.name, e))
             console.print(f"  [yellow]⚠[/yellow] {sync_provider.emoji} {sync_provider.name}: [red]{e}[/red]")
 
+    # Index dbt projects if repos were synced
+    _index_dbt_projects(project_path, results)
```
Comment (Bl3f) [inline cli/nao_core/commands/sync/__init__.py]: I don't think this should be called from there, I'd say as it's related to repos provider that it's called from there as the sync method here is kinda generic.

File: cli/nao_core/dbt_indexer.py [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+    models_dir = project_path / "models"
+    if not models_dir.is_dir():
+        return [], []
+
+    models: list[ModelInfo] = []
+    sources: list[SourceInfo] = []
+    descriptions: dict[str, str] = {}
+
+    # First pass: collect YAML descriptions and sources
+    for yaml_path in models_dir.rglob("*.yml"):
```
Comment (Bl3f) [inline cli/nao_core/dbt_indexer.py]: Feel like this code is repeated twice.

File: cli/nao_core/dbt_indexer.py [hunk 4]
```diff
@@ -0,0 +1,473 @@
...
+def find_dbt_projects(repos_path: Path) -> list[tuple[str, Path]]:
+    """Return (repo_name, dbt_project_dir) for each repo with a dbt project."""
+    results: list[tuple[str, Path]] = []
+
+    for entry in sorted(repos_path.iterdir()):
+        if not entry.is_dir():
+            continue
+
+        root_project = entry / "dbt_project.yml"
+        nested_project = entry / "dbt" / "dbt_project.yml"
```
Comment (Bl3f) [inline cli/nao_core/dbt_indexer.py]: There are cases I think where the nested folder is not named dbt

File: cli/nao_core/dbt_indexer.py [hunk 5]
```diff
@@ -0,0 +1,473 @@
...
+            descriptions.update(parse_yaml_descriptions(yaml_path))
+        except Exception as e:
+            print(f"[dbt-indexer] Warning: failed to parse descriptions from {yaml_path}: {e}")
+        try:
+            sources.extend(parse_yaml_sources(yaml_path))
+        except Exception as e:
+            print(f"[dbt-indexer] Warning: failed to parse sources from {yaml_path}: {e}")
+
+    # Second pass: parse SQL models
+    for sql_path in models_dir.rglob("*.sql"):
```
Comment (Bl3f) [inline cli/nao_core/dbt_indexer.py:150]: I think the 3 forloops here could be factorised and make this code way easier to read.

File: apps/backend/src/agents/user-rules.ts [hunk 6]
```diff
@@ -28,6 +28,58 @@ export function getUserRules(): string | null {
...
+	try {
+		const entries = readdirSync(reposPath, { withFileTypes: true });
+		const repositories: Repository[] = [];
+
+		for (const entry of entries) {
+			if (!entry.isDirectory()) {
+				continue;
+			}
+
+			const rootDbtProject = join(reposPath, entry.name, 'dbt_project.yml');
```
Comment (Bl3f) [inline apps/backend/src/agents/user-rules.ts:59]: The whole search dbt_project.yml should be factorised in a search function (maybe we can use the search tool from the backend).

File: cli/nao_core/dbt_indexer.py [hunk 7]
```diff
@@ -0,0 +1,473 @@
...
+    manifest = generate_manifest_md(models, repo_name, rel_project_path, project_name)
+    (out_dir / "manifest.md").write_text(manifest, encoding="utf-8")
+
+    sources_md = generate_sources_md(sources, repo_name)
+    (out_dir / "sources.md").write_text(sources_md, encoding="utf-8")
+
+    print(f"[dbt-indexer] Indexed {repo_name}: {len(models)} models, {len(sources)} sources")
+
+
+# ---------------------------------------------------------------------------
```
Comment (Bl3f) [inline cli/nao_core/dbt_indexer.py]: All these comments can be removed, the function names are enough.

File: apps/backend/src/agents/user-rules.ts [hunk 8]
```diff
@@ -28,6 +28,58 @@ export function getUserRules(): string | null {
...
 }
 
+type Repository = {
+	name: string;
+	hasDbtProject: boolean;
+	dbtProjectPath?: string;
+};
+
+export function getRepositories(): Repository[] | null {
+	const projectFolder = env.NAO_DEFAULT_PROJECT_PATH;
```
Comment (Bl3f) [inline apps/backend/src/agents/user-rules.ts:38]: This is legacy and should not be used (I know the getConnections does it but it shouldn't), we should be given the project information from the agent when building the system-prompt - this way we are more functional and creating the needed project isolation.

File: cli/nao_core/dbt_indexer.py [hunk 9]
```diff
@@ -0,0 +1,473 @@
...
+    descriptions: dict[str, str] = {}
+
+    # First pass: collect YAML descriptions and sources
+    for yaml_path in models_dir.rglob("*.yml"):
+        if _should_skip(yaml_path) or not yaml_path.is_file():
+            continue
+        try:
+            descriptions.update(parse_yaml_descriptions(yaml_path))
+        except Exception as e:
+            print(f"[dbt-indexer] Warning: failed to parse descriptions from {yaml_path}: {e}")
```
Comment (Bl3f) [inline cli/nao_core/dbt_indexer.py]: Use UI. (ui.py) methods to print in the console.

File: cli/nao_core/commands/sync/__init__.py [hunk 10]
```diff
@@ -8,6 +8,7 @@
 from rich.console import Console
 
 from nao_core.config import NaoConfig
+from nao_core.dbt_indexer import index_all_projects
```
Comment (Bl3f) [inline cli/nao_core/commands/sync/__init__.py]: Maybe we should add in the sync command an option to deactivate this indexing behaviour (or might it be an option on the repo configuration in nao_config.yaml. Like:

is_dbt_indexed: True | False

File: apps/backend/src/components/system-prompt.tsx [hunk 11]
```diff
@@ -69,6 +69,19 @@ export function SystemPrompt() {
 					Each table have files describing the table schema and the data in the table (like columns.md,
 					preview.md, etc.)
 				</ListItem>
+				<ListItem>
+					The <Italic>repos</Italic> folder contains synced git repositories with source code, data
```
Comment (Bl3f) [inline apps/backend/src/components/system-prompt.tsx:75]: dbt mention here should be conditional to the fact that we have a dbt project because if not it will make the agent believe we are in a dbt context.

File: apps/backend/src/agents/user-rules.ts [hunk 12]
```diff
@@ -28,6 +28,58 @@ export function getUserRules(): string | null {
...
+
+			let hasDbtProject = false;
+			let dbtProjectPath: string | undefined;
+
+			if (existsSync(rootDbtProject)) {
+				hasDbtProject = true;
+				dbtProjectPath = `repos/${entry.name}`;
+			} else if (existsSync(subDbtProject)) {
+				hasDbtProject = true;
+				dbtProjectPath = `repos/${entry.name}/dbt`;
```
Comment (Bl3f) [inline apps/backend/src/agents/user-rules.ts:70]: Not enough generic here as well.

File: cli/nao_core/dbt_indexer.py [hunk 13]
```diff
@@ -0,0 +1,473 @@
...
+        "",
+        f"- **Project name:** {project_name}",
+        f"- **dbt project path:** {project_path}",
+        f"- **Total models:** {len(models)}",
+        f"- **Indexed at:** {now}",
+        "",
+        "---",
+    ]
+
+    for model in models:
```
Comment (Bl3f) [inline cli/nao_core/dbt_indexer.py:373]: I'm a bit concerned about the size of the generated manifest file in the end. You said you ran it on a ~700 models dbt project, the issue is that each model entry is roughly ~60 tokens, which in the end lead to a file with 37k characters - if the LLM decide to read it and add it to the context it will overflow the context window very fast.

Maybe we can think of a better structure (and not a single file to avoid this case), on the other we also have to work on a better system prompt for this, the fact that files can get out of the context window and read using a range (#193 PR does it)

File: apps/backend/fastapi/main.py [hunk 14]
```diff
@@ -153,6 +171,7 @@ async def refresh_context():
         updated = provider.refresh()
 
         if updated:
+            _run_dbt_indexing("Refresh")
```
Comment (Bl3f) [inline apps/backend/fastapi/main.py]: As you move this to the repo provider I think it's not needed anymore to have the refresh done here in this file.

File: cli/nao_core/dbt_indexer.py [hunk 15]
```diff
@@ -0,0 +1,473 @@
...
+    """Scan repos/, find dbt projects, index each, write to dbt-index/."""
+    repos_path = project_folder / "repos"
+    if not repos_path.is_dir():
+        return
+
+    projects = find_dbt_projects(repos_path)
+    if not projects:
+        return
+
+    index_root = project_folder / "dbt-index"
```
Comment (Bl3f) [inline cli/nao_core/dbt_indexer.py:63]: I don't know if dbt-index is a great name, what about dbt-projects, dbt-compiled or even without an hyphen which is usually not so good for folder names?

File: apps/backend/src/components/system-prompt.tsx [hunk 16]
```diff
@@ -69,6 +69,19 @@ export function SystemPrompt() {
...
+					transformation logic (dbt models), and documentation.
+				</ListItem>
+				<ListItem>
```
Comment (Bl3f) [inline apps/backend/src/components/system-prompt.tsx]: This should be conditioned because some users will not have the dbt-index, nor any dbt project.

File: cli/nao_core/dbt_indexer.py (hunk 1 above)
Comment (ealexisaraujo) [reply cli/nao_core/dbt_indexer.py:35]: Good point. Tested on a real 694-model project and it correctly extracted all standard `ref()` and `source()` calls. You're right that macro-generated refs (`{{ generate_join(...) }}`) won't be captured — that's an inherent limitation of static analysis without running dbt. For those cases, PR #213's compiled approach would give complete lineage. The regex covers the ~95% common case and works without requiring dbt setup.

Open to ideas on how to handle the gap — maybe a note in the manifest header flagging that macro-generated refs aren't captured?

File: cli/nao_core/commands/sync/__init__.py (hunk 2 above)
Comment (ealexisaraujo) [reply cli/nao_core/commands/sync/__init__.py]: Done — moved to `RepositorySyncProvider.sync()` as a post-sync step. The generic sync no longer knows about dbt indexing.

File: cli/nao_core/dbt_indexer.py (hunk 3 above)
Comment (ealexisaraujo) [reply cli/nao_core/dbt_indexer.py]: Done — factorized into a single loop using `itertools.chain(models_dir.rglob("*.yml"), models_dir.rglob("*.yaml"))`.

File: cli/nao_core/dbt_indexer.py (hunk 4 above)
Comment (ealexisaraujo) [reply cli/nao_core/dbt_indexer.py]: Good catch — replaced with `_find_dbt_project_yml()` that searches up to 3 levels deep via glob patterns, finding `dbt_project.yml` regardless of the parent directory name.

File: cli/nao_core/dbt_indexer.py (hunk 5 above)
Comment (ealexisaraujo) [reply cli/nao_core/dbt_indexer.py:150]: The YAML loops are now 1 loop (via `chain`). The SQL loop stays separate since it depends on descriptions collected from the YAML pass — these are sequential by design (YAML first collects descriptions, then SQL uses them).

File: apps/backend/src/agents/user-rules.ts (hunk 6 above)
Comment (ealexisaraujo) [reply apps/backend/src/agents/user-rules.ts:59]: Acknowledged — this ties into the broader refactor of `getRepositories()` toward injected project context. Happy to extract a shared search function in a follow-up once the architecture direction is clearer.

File: cli/nao_core/dbt_indexer.py (hunk 7 above)
Comment (ealexisaraujo) [reply cli/nao_core/dbt_indexer.py]: Done — removed all separator blocks and section headers.

File: apps/backend/src/agents/user-rules.ts (hunk 8 above)
Comment (ealexisaraujo) [reply apps/backend/src/agents/user-rules.ts:38]: Agreed — both `getRepositories()` and `getConnections()` read the filesystem directly from env vars instead of receiving project context. Kept `getRepositories()` minimally for the `hasDbtProjects` conditional in the system prompt, but the broader refactor toward injected project isolation makes total sense. Happy to contribute to that in a follow-up — would be great to align on the target architecture for how project context flows into the system prompt.

File: cli/nao_core/dbt_indexer.py (hunk 9 above)
Comment (ealexisaraujo) [reply cli/nao_core/dbt_indexer.py]: Done — all `print()` calls replaced with `UI.success()`, `UI.warn()`, `UI.error()`.

File: cli/nao_core/commands/sync/__init__.py (hunk 10 above)
Comment (ealexisaraujo) [reply cli/nao_core/commands/sync/__init__.py]: Good idea. I think this fits well as a follow-up once the core indexer is stable — an `index_dbt: bool` flag in repo config or a CLI flag like `nao sync --skip-indexing`. It would also pair nicely with PR #213's `compile_dbt_docs` flag, giving users fine-grained control over what happens during sync.

Open to implementing this in a follow-up PR if you'd like.

File: apps/backend/src/components/system-prompt.tsx (hunk 11 above)
Comment (ealexisaraujo) [reply apps/backend/src/components/system-prompt.tsx:75]: Done — dbt-specific instructions are now wrapped in `{hasDbtProjects && (...)}`, computed from `getRepositories()`. Users without dbt projects won't see these instructions, so the agent won't incorrectly assume dbt context.

File: apps/backend/src/agents/user-rules.ts (hunk 12 above)
Comment (ealexisaraujo) [reply apps/backend/src/agents/user-rules.ts:70]: Agreed — this ties into the broader `getRepositories()` refactor toward injected project context. Deferring this to a follow-up since it affects other consumers beyond our PR scope. Happy to contribute to that refactor when the direction is decided.

File: cli/nao_core/dbt_indexer.py (hunk 13 above)
Comment (ealexisaraujo) [reply cli/nao_core/dbt_indexer.py:373]: Great point — this is the key design question. For our test project (694 models), the manifest is ~37k chars (~42k tokens). A few options I see:

1. **Single file + grep + range reads** — if PR #193 lands, the agent can grep for a model name then read just that section. The current H3 headers (`### model_name`) are already grep-friendly. This keeps things simple and composable.
2. **Split by directory** — one manifest file per `models/` subdirectory (e.g., `staging/manifest.md`, `marts/manifest.md`). Smaller files, but more to grep across.
3. **Table-of-contents header** — a lightweight index at the top mapping model names to line numbers, so the agent reads only the TOC first and targets specific ranges.

I lean toward option 1 since it's the simplest and leverages existing infrastructure, but open to whatever works best for the overall system. What's your preferred direction?

File: apps/backend/fastapi/main.py (hunk 14 above)
Comment (ealexisaraujo) [reply apps/backend/fastapi/main.py]: Removed entirely. No dbt indexing hooks in FastAPI anymore — indexing happens exclusively via `nao sync` through the `RepositorySyncProvider`.

File: cli/nao_core/dbt_indexer.py (hunk 15 above)
Comment (ealexisaraujo) [reply cli/nao_core/dbt_indexer.py:63]: Open to renaming. `dbt_projects` (underscore, no hyphen) would be more consistent with Python conventions. `dbt_index` or `dbt_compiled` are also options. What's your preference? Happy to adjust to whatever naming convention fits the project best.

File: apps/backend/src/components/system-prompt.tsx (hunk 16 above)
Comment (ealexisaraujo) [reply apps/backend/src/components/system-prompt.tsx]: Done — this is now conditional on `hasDbtProjects`, same fix as the other dbt mention. Users without dbt projects won't see dbt-index related instructions.

File: cli/nao_core/dbt_indexer.py (hunk 1 above)
Comment (Bl3f) [reply cli/nao_core/dbt_indexer.py:35]: Yeah, mainly good for now to do it static this way.

File: cli/nao_core/commands/sync/__init__.py (hunk 10 above)
Comment (Bl3f) [reply cli/nao_core/commands/sync/__init__.py]: I think the best would be an option in the yaml like the compile_dbt_docs, tho naming should be uniform between the 2 flags. I don't think it should be in the CLI sync option at main level nonetheless.

**Comment by Bl3f** [comment]
Last 2 changes (if I'm not mistaken) to me are:
- add the config in the nao_config.yaml, I want users to be able to deactivate this behaviour
- name the created folder `dbt/`
